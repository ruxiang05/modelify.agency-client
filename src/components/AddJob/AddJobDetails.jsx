import React from 'react';
import PropTypes from 'prop-types';
import FormInput from '../FormInput';
import PageHeader from '../PageHeader';

class AddJobDetails extends React.Component {
  constructor(props) {
    super(props);

    this.saveAndContinue = this.saveAndContinue.bind(this);
  }

  saveAndContinue(event) {
    const { nextStep } = this.props;
    event.preventDefault();
    nextStep();
  }

  render() {
    const {
      title, date, address, pay, description, handleChange,
    } = this.props;
    return (
      <div className="page">
        <PageHeader title="Add Job Details" />
        <form>
          <FormInput type="text" value={title} name="title" title="Title" placeholder="Enter job short descritpion" handleChange={handleChange} />
          <FormInput type="date" value={date} name="date" title="Date" placeholder="Enter job date" handleChange={handleChange} />
          <FormInput type="text" value={address} name="address" title="Address" placeholder="Enter job address" handleChange={handleChange} />
          <FormInput type="number" value={pay} name="pay" title="Payment" placeholder="Enter payment amount" handleChange={handleChange} />
          <FormInput type="textarea" value={description} name="description" title="Description" placeholder="Enter job details" handleChange={handleChange} />
          <button type="submit" onClick={this.saveAndContinue}> Save and Continue </button>
        </form>
      </div>
    );
  }
}

AddJobDetails.propTypes = {
  title: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired,
  address: PropTypes.string.isRequired,
  pay: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired,
};

export default AddJobDetails;
