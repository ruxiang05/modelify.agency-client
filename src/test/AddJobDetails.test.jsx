/* Uses methods and/or components from react and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import AddJobDetails from '../components/AddJob/AddJobDetails';

let wrapper;

const values = {
  model: '',
  title: '',
  date: '2019-04-08',
  address: '',
  pay: '',
  description: '',
  handleChange: jest.fn(),
};

const nextStep = jest.fn();

beforeEach(() => {
  wrapper = shallow(<AddJobDetails nextStep={nextStep} {...values} />);
});

describe('AddJobDetails', () => {
  it('should render AddJobDetails', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should go to next step', () => {
    wrapper
      .find('.form-submit input')
      .first()
      .simulate('click', { preventDefault() {} });
    expect(nextStep).toHaveBeenCalled();
  });
});
