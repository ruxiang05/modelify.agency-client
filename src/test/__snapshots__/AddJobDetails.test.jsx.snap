// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddJobDetails should render AddJobDetails 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddJobDetails
    address=""
    date="2019-04-08"
    description=""
    handleChange={[MockFunction]}
    model=""
    nextStep={[MockFunction]}
    pay=""
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PageHeader
          backButton={false}
          title="Add Job Details"
        />,
        <div
          className="page"
        >
          <form>
            <FormInput
              handleChange={[MockFunction]}
              name="title"
              placeholder="Enter job short descritpion"
              title="Title"
              type="text"
              value=""
            />
            <FormInput
              handleChange={[MockFunction]}
              name="date"
              placeholder="Enter job date"
              title="Date"
              type="date"
              value="2019-04-08"
            />
            <FormInput
              handleChange={[MockFunction]}
              name="address"
              placeholder="Enter job address"
              title="Address"
              type="text"
              value=""
            />
            <FormInput
              handleChange={[MockFunction]}
              name="pay"
              placeholder="Enter payment amount"
              title="Payment"
              type="number"
              value=""
            />
            <FormInput
              handleChange={[MockFunction]}
              name="description"
              placeholder="Enter job details"
              title="Description"
              type="textarea"
              value=""
            />
            <div
              className="form-submit"
            >
              <input
                onClick={[Function]}
                type="submit"
                value="Save and Continue"
              />
            </div>
          </form>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "action": undefined,
          "actionAltText": undefined,
          "actionIcon": undefined,
          "backButton": false,
          "goBackAction": undefined,
          "history": undefined,
          "title": "Add Job Details",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form>
            <FormInput
              handleChange={[MockFunction]}
              name="title"
              placeholder="Enter job short descritpion"
              title="Title"
              type="text"
              value=""
            />
            <FormInput
              handleChange={[MockFunction]}
              name="date"
              placeholder="Enter job date"
              title="Date"
              type="date"
              value="2019-04-08"
            />
            <FormInput
              handleChange={[MockFunction]}
              name="address"
              placeholder="Enter job address"
              title="Address"
              type="text"
              value=""
            />
            <FormInput
              handleChange={[MockFunction]}
              name="pay"
              placeholder="Enter payment amount"
              title="Payment"
              type="number"
              value=""
            />
            <FormInput
              handleChange={[MockFunction]}
              name="description"
              placeholder="Enter job details"
              title="Description"
              type="textarea"
              value=""
            />
            <div
              className="form-submit"
            >
              <input
                onClick={[Function]}
                type="submit"
                value="Save and Continue"
              />
            </div>
          </form>,
          "className": "page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FormInput
                handleChange={[MockFunction]}
                name="title"
                placeholder="Enter job short descritpion"
                title="Title"
                type="text"
                value=""
              />,
              <FormInput
                handleChange={[MockFunction]}
                name="date"
                placeholder="Enter job date"
                title="Date"
                type="date"
                value="2019-04-08"
              />,
              <FormInput
                handleChange={[MockFunction]}
                name="address"
                placeholder="Enter job address"
                title="Address"
                type="text"
                value=""
              />,
              <FormInput
                handleChange={[MockFunction]}
                name="pay"
                placeholder="Enter payment amount"
                title="Payment"
                type="number"
                value=""
              />,
              <FormInput
                handleChange={[MockFunction]}
                name="description"
                placeholder="Enter job details"
                title="Description"
                type="textarea"
                value=""
              />,
              <div
                className="form-submit"
              >
                <input
                  onClick={[Function]}
                  type="submit"
                  value="Save and Continue"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [MockFunction],
                "name": "title",
                "placeholder": "Enter job short descritpion",
                "title": "Title",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [MockFunction],
                "name": "date",
                "placeholder": "Enter job date",
                "title": "Date",
                "type": "date",
                "value": "2019-04-08",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [MockFunction],
                "name": "address",
                "placeholder": "Enter job address",
                "title": "Address",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [MockFunction],
                "name": "pay",
                "placeholder": "Enter payment amount",
                "title": "Payment",
                "type": "number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [MockFunction],
                "name": "description",
                "placeholder": "Enter job details",
                "title": "Description",
                "type": "textarea",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  onClick={[Function]}
                  type="submit"
                  value="Save and Continue"
                />,
                "className": "form-submit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onClick": [Function],
                  "type": "submit",
                  "value": "Save and Continue",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageHeader
            backButton={false}
            title="Add Job Details"
          />,
          <div
            className="page"
          >
            <form>
              <FormInput
                handleChange={[MockFunction]}
                name="title"
                placeholder="Enter job short descritpion"
                title="Title"
                type="text"
                value=""
              />
              <FormInput
                handleChange={[MockFunction]}
                name="date"
                placeholder="Enter job date"
                title="Date"
                type="date"
                value="2019-04-08"
              />
              <FormInput
                handleChange={[MockFunction]}
                name="address"
                placeholder="Enter job address"
                title="Address"
                type="text"
                value=""
              />
              <FormInput
                handleChange={[MockFunction]}
                name="pay"
                placeholder="Enter payment amount"
                title="Payment"
                type="number"
                value=""
              />
              <FormInput
                handleChange={[MockFunction]}
                name="description"
                placeholder="Enter job details"
                title="Description"
                type="textarea"
                value=""
              />
              <div
                className="form-submit"
              >
                <input
                  onClick={[Function]}
                  type="submit"
                  value="Save and Continue"
                />
              </div>
            </form>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": undefined,
            "actionAltText": undefined,
            "actionIcon": undefined,
            "backButton": false,
            "goBackAction": undefined,
            "history": undefined,
            "title": "Add Job Details",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form>
              <FormInput
                handleChange={[MockFunction]}
                name="title"
                placeholder="Enter job short descritpion"
                title="Title"
                type="text"
                value=""
              />
              <FormInput
                handleChange={[MockFunction]}
                name="date"
                placeholder="Enter job date"
                title="Date"
                type="date"
                value="2019-04-08"
              />
              <FormInput
                handleChange={[MockFunction]}
                name="address"
                placeholder="Enter job address"
                title="Address"
                type="text"
                value=""
              />
              <FormInput
                handleChange={[MockFunction]}
                name="pay"
                placeholder="Enter payment amount"
                title="Payment"
                type="number"
                value=""
              />
              <FormInput
                handleChange={[MockFunction]}
                name="description"
                placeholder="Enter job details"
                title="Description"
                type="textarea"
                value=""
              />
              <div
                className="form-submit"
              >
                <input
                  onClick={[Function]}
                  type="submit"
                  value="Save and Continue"
                />
              </div>
            </form>,
            "className": "page",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FormInput
                  handleChange={[MockFunction]}
                  name="title"
                  placeholder="Enter job short descritpion"
                  title="Title"
                  type="text"
                  value=""
                />,
                <FormInput
                  handleChange={[MockFunction]}
                  name="date"
                  placeholder="Enter job date"
                  title="Date"
                  type="date"
                  value="2019-04-08"
                />,
                <FormInput
                  handleChange={[MockFunction]}
                  name="address"
                  placeholder="Enter job address"
                  title="Address"
                  type="text"
                  value=""
                />,
                <FormInput
                  handleChange={[MockFunction]}
                  name="pay"
                  placeholder="Enter payment amount"
                  title="Payment"
                  type="number"
                  value=""
                />,
                <FormInput
                  handleChange={[MockFunction]}
                  name="description"
                  placeholder="Enter job details"
                  title="Description"
                  type="textarea"
                  value=""
                />,
                <div
                  className="form-submit"
                >
                  <input
                    onClick={[Function]}
                    type="submit"
                    value="Save and Continue"
                  />
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [MockFunction],
                  "name": "title",
                  "placeholder": "Enter job short descritpion",
                  "title": "Title",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [MockFunction],
                  "name": "date",
                  "placeholder": "Enter job date",
                  "title": "Date",
                  "type": "date",
                  "value": "2019-04-08",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [MockFunction],
                  "name": "address",
                  "placeholder": "Enter job address",
                  "title": "Address",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [MockFunction],
                  "name": "pay",
                  "placeholder": "Enter payment amount",
                  "title": "Payment",
                  "type": "number",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [MockFunction],
                  "name": "description",
                  "placeholder": "Enter job details",
                  "title": "Description",
                  "type": "textarea",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    onClick={[Function]}
                    type="submit"
                    value="Save and Continue"
                  />,
                  "className": "form-submit",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "onClick": [Function],
                    "type": "submit",
                    "value": "Save and Continue",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": "div",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
